src/util/ecdsa.rs

```diff
@@ -413,6 +417,61 @@ impl<'de> ::serde::Deserialize<'de> for PublicKey {
     }
 }
 
+/// An ECDSA signature with the corresponding hash type.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
```

Comment:
`#[derive(PartialOrd, Ord, Hash]`

Reply:
What does it mean for signature to be greater than other signature? Doesn't seem intuitive for me. I agree with `Hash`.

Reply:
It means that they can be used as keys in `BTreeMap` or be put into `BTreeSet`. Maps are important for stuff like PSBTs.

Signatures _do_ implement `Ord` in secp library (both in rust and C versions), so this should be followed here as well, I assume

Reply:
For PSBTs and such I'd expect maps to be keyed by the key (:D) not signature. However if this is widely implemented then I agree with adding it as long as ordering works the same as in other sw.

Reply:
PSBTs are indeed keyed not by signatures, but I can well imagine wallets needing to key some data by signatures

Reply:
Only `Ord, PartialOrd` can be auto derived for SchnorrSig. `PartialOrd, Ord, Hash` cannot be derived for `EcdsaSig`. Added `Ord, PartialOrd` for now